contracts/governance/extensions/GovernorSequentialProposalId.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+
+/**
+ * @dev Extension of {Governor} that changes the numbering of proposal ids from the default hash-based approach to
+ * sequential ids.
+ */
+abstract contract GovernorSequentialProposalId is Governor {
+    uint256 private _nextProposalId = 1;
```

Comment:
I'm not a fan of storage variables that have to be initialized in the constructor. This means upgradeable version need an initializer or a re-initializer.

IMO we should use `_lastProposalId` instead. We can then adapt `nextProposalId()` to return `_lastProposalId+1` if we want.